### PR TITLE
Expand Zoho node

### PR DESCRIPTION
**Pull request by ivov**
This PR expands the Zoho node:
```
account
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
contact
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
deal
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
invoice
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
lead
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
product
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
purchase
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
quote
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
salesOrder
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
vendor
  ├── create
  ├── delete
  ├── update
  ├── upsert
  ├── get
  └── getAll
```

### CODE REVIEW

File: packages/nodes-base/nodes/Zoho/ZohoCrm.node.ts [hunk 1]
```diff
@@ -6,37 +6,71 @@ import {
...
 
 export class ZohoCrm implements INodeType {
 	description: INodeTypeDescription = {
-		displayName: 'Zoho CRM',
-		name: 'zohoCrm',
-		icon: 'file:zohoCrm.png',
-		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
-		group: ['input'],
+		displayName: 'Zoho',
+		name: 'zoho',
```
Comment (MedAliMarz) [inline packages/nodes-base/nodes/Zoho/ZohoCrm.node.ts:66]: Changing the name will break existing workflow, maybe delay this small changes after the merge of node versions feature

**Review by MedAliMarz — COMMENTED** [review]
Great work @ivov ; just added few recommendations, and mentioned one minor bug when retrieving multiple resource (in the case of empty result), Thank you

**Review by MedAliMarz — COMMENTED** [review]
Hi @ivov , thanks for applying the suggestions, just found few corner cases to handle, and another suggestion to improve the output of create/update/delete operation (which can be ignored)

File: packages/nodes-base/nodes/Zoho/descriptions/PurchaseOrderDescription.ts [hunk 2]
```diff
@@ -86,11 +91,12 @@ export const purchaseOrderFields = [
 				],
 				operation: [
 					'create',
+					'upsert',
 				],
 			},
 		},
 	},
-	makeProductDetails('purchaseOrder', 'create'),
+	makeProductDetails('purchaseOrder', 'create', { hasUpsert: true }),
```
Comment (MedAliMarz) [inline packages/nodes-base/nodes/Zoho/descriptions/PurchaseOrderDescription.ts:99]: This UI displayOptions misses upsert in the show section (the additional fields are not found by the code).

Comment (MedAliMarz) [inline packages/nodes-base/nodes/Zoho/descriptions/PurchaseOrderDescription.ts:99]: The additional fields UI element ,misses `upsert` in show section (displayOptions)

**Review by MedAliMarz — COMMENTED** [review]
Only small issue related to upsert in the purchaseOrder, also the other getAll operation doesn't include the filters, maybe adding them for better clarity?

File: packages/nodes-base/nodes/Zoho/ZohoCrm.node.ts [hunk 3]
```diff
@@ -1202,7 +1290,12 @@ export class ZohoCrm implements INodeType {
 						//            vendor: getAll
 						// ----------------------------------------
 
-						responseData = await handleListing.call(this, 'GET', '/vendors');
+						const qs: IDataObject = {};
+						const options = this.getNodeParameter('options', i) as GetAllFilterOptions;
+
+						addGetAllFilterOptions(qs, options);
+
+						responseData = await handleListing.call(this, 'GET', '/sales_orders');
```
Comment (MedAliMarz) [inline packages/nodes-base/nodes/Zoho/ZohoCrm.node.ts:1347]: I think this should be '/vendors'. and missing the querystring

File: packages/nodes-base/nodes/Zoho/GenericFunctions.ts [hunk 4]
```diff
@@ -225,44 +226,71 @@ const adjustAccountIdField = adjustIdField('accountId', 'Account_Name');
 const adjustContactIdField = adjustIdField('contactId', 'Full_Name');
 const adjustDealIdField = adjustIdField('dealId', 'Deal_Name');
 
+const adjustCustomFields = (allFields: AllFields) => {
```
Comment (MedAliMarz) [inline packages/nodes-base/nodes/Zoho/GenericFunctions.ts:228]: This function isn't called for `product` resource, that's why the custom field doesn't get added to the record.

**Review by MedAliMarz — COMMENTED** [review]
Added a few comments regarding the filter options and the custom fields.

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with n8n@0.127.0

**Pull request merged**